build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/release/ChangelogEntry.java

```diff
@@ -36,7 +36,7 @@
     private String summary;
     private Highlight highlight;
     private Breaking breaking;
-    private Deprecation deprecation;
+    private Breaking deprecation;
```

Comment:
Are we just consolidating types because they're essentially the same thing here? If so the naming is a bit confusing. Maybe something with "compatibility" in the name or just declaring a `Deprecation` class the just extends `Breaking` (a type alias effectively).

Reply:
Fair point, I've renamed `Breaking` and made it abstract, and added `Breaking` and `Deprecation` as aliases.